bin/ch/Helpers.cpp

```diff
@@ -16,244 +16,92 @@
-    size_t dstRemainingByteLength = (dstLength - dpos - 1) * sizeof(TTDHostCharType);
-    Helpers::TTReportLastIOErrorAsNeeded(srcByteLength <= dstRemainingByteLength, "The source string must be able to fit within the destination buffer");
-
-    memcpy_s(dst + dpos, dstRemainingByteLength, src, srcByteLength);
-    dst[dpos + srcLength] = '\0';
-}
-
-void TTDHostAppendWChar(TTDHostCharType* dst, size_t dstLength, const wchar* src)
-{
-    size_t srcLength = wcslen(src);
-    size_t dpos = TTDHostStringLength(dst);
-    Helpers::TTReportLastIOErrorAsNeeded(dpos < dstLength, "The end of the string already exceeds the buffer");
-
-    size_t dstRemainingLength = dstLength - dpos - 1;
-    Helpers::TTReportLastIOErrorAsNeeded(srcLength <= dstRemainingLength, "The source string must be able to fit within the destination buffer");
-
-    // TODO - analyze this function further
-    utf8::EncodeIntoAndNullTerminate(dst + dpos, src, srcLength);
-}
-
-void TTDHostAppendAscii(TTDHostCharType* dst, size_t dstLength, const char* src)
+void TTDHostBuildCurrentExeDirectory(char* path, size_t* pathLength)
 {
-    size_t srcLength = strlen(src);
-    size_t dpos = TTDHostStringLength(dst);
-    Helpers::TTReportLastIOErrorAsNeeded(dpos < dstLength, "The end of the string already exceeds the buffer");
-
-    size_t srcByteLength = srcLength * sizeof(TTDHostCharType);
-    size_t dstRemainingByteLength = (dstLength - dpos - 1) * sizeof(TTDHostCharType);
-    Helpers::TTReportLastIOErrorAsNeeded(srcByteLength <= dstRemainingByteLength, "The source string must be able to fit within the destination buffer");
-
-    memcpy_s(dst + dpos, dstRemainingByteLength, src, srcByteLength);
-    dst[dpos + srcLength] = '\0';
-}
-
-void TTDHostBuildCurrentExeDirectory(TTDHostCharType* path, size_t pathBufferLength)
-{
-    TTDHostCharType exePath[MAX_PATH];
+    char exePath[MAX_URI_LENGTH];
     //TODO: xplattodo move this logic to PAL
     #ifdef __APPLE__
     uint32_t tmpPathSize = sizeof(exePath);
-    _NSGetExecutablePath(TTDHostCharConvert(exePath), &tmpPathSize);
-    size_t len = TTDHostStringLength(exePath);
+    _NSGetExecutablePath(exePath, &tmpPathSize);
+    size_t i = strlen(exePath) - 1;
     #else
-    size_t len = readlink("/proc/self/exe", TTDHostCharConvert(exePath), MAX_PATH);
+    size_t i = readlink("/proc/self/exe", exePath, MAX_URI_LENGTH) - 1;
     #endif
 
-    size_t i = len - 1;
-    while(exePath[i] != TTDHostPathSeparatorChar)
+    while(exePath[i] != '/')
     {
         --i;
     }
+    *pathLength = i + 1;
 
-    exePath[i + 1] = '\0';
-
-    TTDHostAppend(path, pathBufferLength, exePath);
+    memcpy_s(path, MAX_URI_LENGTH, exePath, *pathLength);
 }
 
-JsTTDStreamHandle TTDHostOpen(const TTDHostCharType* path, bool isWrite)
+int TTDHostMKDir(const char* path, size_t pathLength)
 {
-    return (JsTTDStreamHandle)fopen(TTDHostCharConvert(path), isWrite ? "w+b" : "r+b");
+    return mkdir(path, 0777);
```

Comment:
Do we want to give an exec. rights? otherwise, folder access should be `0666` instead of `0777`.  Actually reader and writer are the same users, even `0600` should be fine.